src/openms/include/OpenMS/METADATA/ExperimentalDesign.h

```diff
@@ -161,21 +168,51 @@ namespace OpenMS
 
     void setSampleSection(const SampleSection& sample_section);
 
+    /// returns a map from a sample section row to sample id to cluster
+    /// duplicate sample rows (e.g. to find all fractions of the same "sample")
+    std::map<std::vector<String>, std::set<unsigned>> getUniqueSampleRowToSampleMapping() const;
+
+    /// uses getUniqueSampleRowToSampleMapping to get the reversed map
+    /// mapping sample ID to a real unique sample
+    std::map<unsigned, unsigned> getSampleToPrefractionationMapping() const;
+
     /// return fraction index to file paths (ordered by fraction_group)
+    //TODO this probably needs a basename parameter to be fully compatible with the other mappings!! Implicit full path.
     std::map<unsigned int, std::vector<String> > getFractionToMSFilesMapping() const;
 
+    /// return vector of filepath/label combinations that share the same conditions after removing
+    /// replicate columns in the sample section (e.g. for merging across replicates)
+    //TODO this probably needs a basename parameter to be fully compatible with the other mappings!! Implicit full path.
+    std::vector<std::vector<std::pair<String, unsigned>>> getConditionToPathLabelVector() const;
+
+    /// return a condition (unique combination of sample section values except replicate) to Sample index mapping
+    std::map<std::vector<String>, std::set<unsigned>> getConditionToSampleMapping() const;
+
    /*
     *   The (Path, Label) tuples in the experimental design have to be unique, so we can map them
     *   uniquely to the sample number, fraction number, and fraction_group number
     */
+
+    /// return <file_path, label> to prefractionation mapping (a prefractionation group is a unique combination of
+    /// all columns in the sample section, except for replicates.
+    std::map< std::pair< String, unsigned >, unsigned> getPathLabelToPrefractionationMapping(bool use_basename_only) const;
```

Comment:
please use typedefs for this (or small structs) to make it more clear what is going on.